plugins/module_utils/idrac_redfish.py

```diff
@@ -180,7 +183,7 @@ def invoke_request(self, uri, method, data=None, query_param=None, headers=None,
 
     def __enter__(self):
         """Creates sessions by passing it to header"""
-        if self.req_session:
+        if self.req_session and not self.x_auth_token:
```

Comment:
Make it much easier to understand like below:

if self.req_session:
     if self.x_auth_token:
           taken token in header
     else:
           take username and password in payload